service/src/main.rs

```diff
@@ -621,52 +622,12 @@ fn subscribe_to_parentchain_new_headers<E: EnclaveBase + Sidechain>(
 			new_header.number
 		);
 
-		last_synced_header = sync_parentchain(enclave_api, api, last_synced_header);
+		let parentchain_block_syncer =
+			ParentchainBlockSyncer::new(api.clone(), (*enclave_api).clone());
```

Comment:
So you said that you already tried passing an `Arc<EnclaveApi>` with `enclave.clone()` here?

Reply:
Now I managed to do that and it seems to work like that!